test/test_magnet.cpp

```diff
@@ -476,3 +476,82 @@ TORRENT_TEST(invalid_web_seed_escaping)
 	TEST_CHECK(ec);
 }
 
+TORRENT_TEST(parse_magnet_select_only)
+{
+	error_code ec;
+	add_torrent_params p = parse_magnet_uri("magnet:?xt=urn:btih:cdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcd"
+		"&dn=foo&so=0,2,4,6-8", ec);
+	TEST_CHECK(!ec);
+
+	TEST_EQUAL(p.file_priorities.size(), 9);
+	for (std::size_t i = 0; i < p.file_priorities.size(); i++)
+	{
+		if (i == 0 || i == 2 || i == 4 || i == 6 || i == 7 || i == 8)
+		{
+			TEST_EQUAL(p.file_priorities[i], default_priority);
+		}
+		else
+		{
+			TEST_EQUAL(p.file_priorities[i], dont_download);
+		}
+	}
+
+	p = parse_magnet_uri("magnet:?xt=urn:btih:cdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcd"
+		"&dn=foo&so=0,2-4,3-5&dht=10.0.0.1:1337", ec);
+	TEST_CHECK(!ec);
+
+	TEST_EQUAL(p.file_priorities.size(), 6);
+	for (std::size_t i = 0; i < p.file_priorities.size(); i++)
+	{
+		if (i == 0 || i == 2 || i == 3 || i == 4 || i == 5)
+		{
+			TEST_EQUAL(p.file_priorities[i], default_priority);
+		}
+		else
+		{
+			TEST_EQUAL(p.file_priorities[i], dont_download);
+		}
+	}
+
+	p = parse_magnet_uri("magnet:?xt=urn:btih:cdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcd"
```

Comment:
does the BEP defined multiple arguments being additive?

Reply:
no, that was my interpretation, it could be a reset, you tell me

Reply:
I don't have an opinion. perhaps it makes sense to bring it back to bittorrent.org and see if anyone else does

Reply:
@dessalines what is your position here?